Tests/tests/test_Telemetry.cpp

```diff
@@ -234,3 +251,335 @@ TEST_F(TelemetryRfcTest, Plugin)
+            }));
+
+    EXPECT_CALL(service, Submit(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](const uint32_t, const Core::ProxyType<Core::JSON::IElement>& json) {
+                string text;
+                EXPECT_TRUE(json->ToString(text));
+
+                EXPECT_EQ(text, string(_T(
+                    "{\"jsonrpc\":\"2.0\",\"method\":\"org.rdk.Telemetry.onReportUpload\",\"params\":{\"telemetryUploadStatus\":\"UPLOAD_FAILURE\"}}"
+                )));
+
+                onReportUpload.SetEvent();
+
+                return Core::ERROR_NONE;
+            }));
+
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+
+    handler.Subscribe(0, _T("onReportUpload"), _T("org.rdk.Telemetry"), message);
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("uploadReport"), _T("{}"), response));
+    EXPECT_EQ(Core::ERROR_NONE, onReportUpload.Lock());
+
+    handler.Unsubscribe(0, _T("onReportUpload"), _T("org.rdk.Telemetry"), message);
+
+    plugin->Deinitialize(nullptr);
+}
+
+TEST_F(TelemetryRBusTest, uploadLogs)
+{
+    Core::Event onReportUpload(false, true);
+
+    struct _rbusObject rbObject;
+    struct _rbusValue rbValue;
+
+    EXPECT_CALL(rBusApiImplMock, rbus_open(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](rbusHandle_t* handle, char const* componentName) {
+                EXPECT_TRUE(nullptr != handle);
+                EXPECT_EQ(string(componentName), _T("TelemetryThunderPlugin"));
+                return RBUS_ERROR_SUCCESS;
+            }));
+
+
+    EXPECT_CALL(rBusApiImplMock, rbusValue_GetString(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](rbusValue_t value, int* len) {
+                EXPECT_EQ(value, &rbValue);
+                return "SUCCESS";
+            }));
+
+    EXPECT_CALL(rBusApiImplMock, rbusObject_GetValue(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](rbusObject_t object, char const* name) {
+                EXPECT_EQ(object, &rbObject);
+                EXPECT_EQ(string(name), _T("UPLOAD_STATUS"));
+                return &rbValue;
+            }));
+
+    EXPECT_CALL(rBusApiImplMock, rbusMethod_InvokeAsync(::testing::_, ::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](rbusHandle_t handle, char const* methodName, rbusObject_t inParams, rbusMethodAsyncRespHandler_t callback,  int timeout) {
+                callback(handle, methodName, RBUS_ERROR_SUCCESS, &rbObject);
+
+                return RBUS_ERROR_SUCCESS;
+            }));
+
+    EXPECT_CALL(rBusApiImplMock, rbus_close(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](rbusHandle_t handle) {
+                return RBUS_ERROR_SUCCESS;
+            }));
```

Comment:
please don't [over-specify](https://github.com/rdkcentral/rdkservices/tree/sprint/22Q4/Tests#faq). `callback` is called, `onReportUpload` is checked so the test works. `rbus_open/rbusValue_GetString/rbusObject_GetValue/rbusMethod_InvokeAsync/rbus_close` calls are optional so can be `ON_CALL` instead of `EXPECT_CALL`. Also, if `::testing::Invoke` only returns, please use `::testing::Return`